axlearn/common/trainer_config_modifier.py

```diff
@@ -146,6 +186,111 @@ def __call__(self, cfg: SpmdTrainer.Config) -> SpmdTrainer.Config:
         return cfg
 
 
+class ModelConfigModifier(ConfigModifier):
+    """Update the model config for the trainer config."""
+
+    @config_class
+    class Config(ConfigModifier.Config):
+        """Configure ModelConfigModifier.
+
+        Attributes:
+            model_cfg_modifications: A mapping from module path
+                (e.g. `model.decoder.transformer.layer`) to a Config.
+        """
+
+        model_cfg_modifications: Required[Dict[str, ConfigBase]] = REQUIRED
```

Comment:
How about:

```suggestion
        model_cfg_modifications: Dict[str, Callable[[ConfigBase], ConfigBase]] = {}
```

?

That is, the values are config transformer functions.